radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/SystemConstraintScrypt.java

```diff
@@ -48,6 +55,28 @@ public SystemConstraintScrypt(Set<String> systemNames) {
 		this.systemNames = systemNames;
 	}
 
+	private static class AllocatingSystem implements ReducerState {
+	}
+
+	private static class AllocatingVirtualState implements ReducerState {
+		private final LinkedList<SubstateTypeId> substatesToVirtualize = new LinkedList<>();
+		AllocatingVirtualState() {
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_STAKE_DATA);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_ALLOW_DELEGATION_FLAG);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_REGISTERED_FLAG_COPY);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_RAKE_COPY);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_OWNER_COPY);
```

Comment:
Are we missing `UNCLAIMED_READDRESS`?

Reply:
No, this is handled in a separate procedure